### PR TITLE
Updates to Configuration Section for the hyperopt refactoring for Ray Tune

### PR DESCRIPTION
**Pull request by jimthompson5802**
Updated Configuration Section:

* Remove references to legacy Ludwig hyperopt functions
* Improved write-up for use of Ray Tune capabilities 

In support of PR https://github.com/ludwig-ai/ludwig/pull/1994

### CODE REVIEW

File: docs/configuration/hyperparameter_optimization.md [hunk 1]
```diff
@@ -30,145 +33,86 @@ hyperopt:
...
-    low: 0.001
-    high: 0.1
-  steps: 4
+  space: loguniform
+  lower: 0.001
+  uppper: 0.1
 ```
 
-Integer example:
+**Integer example**: Uniform random integer values 1, 2, 3
```
Comment (w4nderlust) [inline docs/configuration/hyperparameter_optimization.md:67]: It seems that for different spaces upper can be inclusive or exclusive. As it is weird that the behavior changes maybe adding a note (that renders as a note) suggesting checking in the ray documentation if ranges are inclusive or exclusive would be valuable

File: docs/configuration/hyperparameter_optimization.md [hunk 2]
```diff
@@ -30,145 +33,86 @@ hyperopt:
...
 
 Example:
 
 ```yaml
-sampler:
-  type: random
-  num_samples: 10
+text.cell_type:
+  categories: [rnn, gru, lstm]
+  space: choice
```
Comment (w4nderlust) [inline docs/configuration/hyperparameter_optimization.md:97]: I sould always pu space first, just for consistency

File: docs/configuration/hyperparameter_optimization.md [hunk 3]
```diff
@@ -230,6 +178,10 @@ executor:
   gpu_resources_per_trial: 1
   kubernetes_namespace: ray
   time_budget_s: 7200
+  scheduler:
+    type: async_hyperband
+    time_attr: training_iteration
+    reduction_factor: 4
```
Comment (w4nderlust) [inline docs/configuration/hyperparameter_optimization.md:184]: there's a weird indentation here

File: docs/configuration/hyperparameter_optimization.md [hunk 4]
```diff
@@ -193,26 +137,24 @@ hyperopt:
...
+You can find the full list of supported search algorithm names in Ray Tune's [create_searcher](https://github.com/ray-project/ray/blob/master/python/ray/tune/suggest/__init__.py) function. Please note these algorithms require installation of additional packages.  As of this version of Ludwig, Ludwig installs the packages for the search algorithm `hyperopt`.  For all other search algorithms, the user is expected to install the required packages.
+
+# Executor
+
 ## Ray Tune Executor
 
-The `ray` executor is used in conjunction with the `ray` sampler to enable [Ray Tune](https://docs.ray.io/en/master/tune/index.html) for distributed hyperopt across a cluster of machines.
+The `ray` executor is used to enable [Ray Tune](https://docs.ray.io/en/master/tune/index.html) for both local and distributed hyperopt across a cluster of machines.
 
 **Parameters:**
```
Comment (w4nderlust) [inline docs/configuration/hyperparameter_optimization.md:159]: Need to add num_samples here and explain the logic wwhen mixing grid and other spaces

File: docs/configuration/hyperparameter_optimization.md (hunk 1 above)
Comment (jimthompson5802) [reply docs/configuration/hyperparameter_optimization.md:67]: done

File: docs/configuration/hyperparameter_optimization.md (hunk 2 above)
Comment (jimthompson5802) [reply docs/configuration/hyperparameter_optimization.md:97]: done.

File: docs/configuration/hyperparameter_optimization.md (hunk 3 above)
Comment (jimthompson5802) [reply docs/configuration/hyperparameter_optimization.md:184]: I think it is correct indentation.  The keywords `time_attr` and `reduction_factor` are parameters for the `async_hyperband` scheduler: https://docs.ray.io/en/latest/tune/api_docs/schedulers.html#asha-tune-schedulers-ashascheduler

I'm leaving this unresolved for now.  If I'm missing something, I may need more help to understand the point.

**Pull request merged**